verifiers/internal/gha/slsaprovenance/v0.2/byob.go

```diff
@@ -0,0 +1,148 @@
+package v02
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa02 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	serrors "github.com/slsa-framework/slsa-verifier/v2/errors"
+
+	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/common"
+)
+
+// byobBuildType is the base build type for BYOB delegated builders.
+var byobBuildType = "https://github.com/slsa-framework/slsa-github-generator/delegator-generic@v0"
+
+// BYOBProvenanceV02 is SLSA v0.2 provenance for the slsa-github-generator BYOB build type.
+type BYOBProvenanceV02 struct {
+	prov *intotoAttestation
+}
+
+// Predicate implements ProvenanceV02.Predicate.
+func (p *BYOBProvenanceV02) Predicate() slsa02.ProvenancePredicate {
+	return p.prov.Predicate
+}
+
+// BuilderID implements Provenance.BuilderID.
+func (p *BYOBProvenanceV02) BuilderID() (string, error) {
+	return p.prov.Predicate.Builder.ID, nil
+}
+
+// SourceURI implements Provenance.SourceURI.
+func (p *BYOBProvenanceV02) SourceURI() (string, error) {
+	if len(p.prov.Predicate.Materials) == 0 {
```

Comment:
I'm wondering whether our existing code was wrong and the intention was to verify against `sourceConfig`...?
Not very important at this point...